map2loop/config.py

```diff
@@ -175,11 +175,20 @@ def update_from_file(self, filename: str, legacy_format: bool = False, lower: bo
         else:
             func = self.update_from_dictionary
 
-        if filename.startswith("http") or filename.startswith("ftp"):
-            with urllib.request.urlopen(filename) as url_data:
-                data = hjson.load(url_data)
-                func(data, lower)
-        else:
-            with open(filename) as url_data:
-                data = hjson.load(url_data)
-                func(data, lower)
+        try:
+            if filename.startswith("http") or filename.startswith("ftp"):
```

Comment:
Looks good, my only comment is it could be worthwhile having a generic wrapper around urrllib.request and allowing this to catch http exceptions and try again. Its nicer for the code to try a few times rather than throw an exception if it just fails to connect to the server.